01_fib/baijie/fib_test.go

```diff
@@ -0,0 +1,52 @@
+package main
+
+import (
+	"bytes"
+	"strconv"
+	"strings"
+	"testing"
+)
+
+func TestMainOutput(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+
+	main()
+	expected := strconv.Quote("1\n")
+	actual := strconv.Quote(buf.String())
+
+	if expected != actual {
+		t.Errorf("Unexpected output in main(), expected %v, actual %v", expected, actual)
+	}
+}
+
+func TestFib(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+
+	type test struct {
+		input    int
+		expected string
+	}
+
+	s1 := []string{"1", "1", "2", "3", "5", "8", "13"}
+	s2 := []string{"1", "-1", "2", "-3", "5", "-8", "13"}
+
+	tests := []test{
+		{input: 0, expected: strconv.Quote("")},
+		{input: 1, expected: strconv.Quote("1\n")},
+		{input: -1, expected: strconv.Quote("1\n")},
+		{input: 7, expected: strconv.Quote(strings.Join(s1, "\n") + "\n")},
+		{input: -7, expected: strconv.Quote(strings.Join(s2, "\n") + "\n")},
+	}
```

Comment:
While I agree that the strings look much neater like this, pushing everything inside a _anonymous struct literal_ makes it look even better. Have a look at [https://dave.cheney.net/2019/05/07/prefer-table-driven-tests](url).

Reply:
Done